yelp_scrape.py

```diff
@@ -23,16 +24,30 @@ def url_generator(search_item, location):
     search_item = search_item.replace(' ', '+')
     location = location.replace(' ', '+').replace(',', '%2C')
     url = base_url + 'find_desc=' + search_item + '&find_loc=' + location
-    return url
-
-
-search, loc = get_search_request()
-yelp_page = requests.get(url_generator(search, loc))
-yelp_soup = BeautifulSoup(yelp_page.content, 'html.parser')
-
-test = yelp_soup.find_all('a')
-links = []
-for a in test:
-    links.append(a["href"])
-
-print(links)
+    file_name = search_item + location
+    return url, file_name
+
+# local_cache_check will take the search request url and check a local directory for a file of the same name
+# if the file is found then it will use that file instead of scraping the internet
+def local_cache_check(url, file_name):
+    # url_hash = hash(url)
+    path = input('Please enter complete filepath to search: ')
+    file_list = os.listdir(path)
```

Comment:
so you don't need to list all the files, then find the one that matches
you should be able to just do `with open(file_name, 'r') as file:` directly